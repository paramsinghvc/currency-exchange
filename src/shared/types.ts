import { HomeState } from "scenes/Home/home.redux";

export interface IRootState {
  home: HomeState;
}

export interface IObject<T = any> {
  [k: string]: T;
}
