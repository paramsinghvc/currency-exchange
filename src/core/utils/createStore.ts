import { applyMiddleware, createStore } from "redux";
import createSagaMiddleware from "redux-saga";

import rootReducer from "./rootReducer";
import { configureSaga } from "./rootSaga";
import logger from "./loggerMiddleware";

export function getPreviousState() {
  const store = localStorage.getItem("store");
  if (store) {
    try {
      const parsedStore = JSON.parse(store);
      return parsedStore;
    } catch (e) {
      return undefined;
    }
  }
  return undefined;
}

export default (preloadedState = getPreviousState()) => {
  const sagaMiddleware = createSagaMiddleware();
  const middlewares = [logger<any>(), sagaMiddleware];
  const middlewareEnhancer = applyMiddleware(...middlewares);

  const store = createStore(rootReducer, preloadedState, middlewareEnhancer);
  sagaMiddleware.run(configureSaga());
  window.onbeforeunload = function() {
    localStorage.setItem("store", JSON.stringify(store.getState()));
  };
  return store;
};
