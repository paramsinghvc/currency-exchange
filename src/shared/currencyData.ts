export default {
  ALL: {
    name: "Albania Lek",
    symbol: "&#76;&#101;&#107;"
  },
  AFN: {
    name: "Afghanistan Afghani",
    symbol: "&#1547;"
  },
  ARS: {
    name: "Argentina Peso",
    symbol: "&#36;"
  },
  AWG: {
    name: "Aruba Guilder",
    symbol: "&#402;"
  },
  AUD: {
    name: "Australia Dollar",
    symbol: "&#36;"
  },
  AZN: {
    name: "Azerbaijan New Manat",
    symbol: "&#1084;&#1072;&#1085;"
  },
  BSD: {
    name: "Bahamas Dollar",
    symbol: "&#36;"
  },
  BBD: {
    name: "Barbados Dollar",
    symbol: "&#36;"
  },
  BYR: {
    name: "Belarus Ruble",
    symbol: "&#112;&#46;"
  },
  BZD: {
    name: "Belize Dollar",
    symbol: "&#66;&#90;&#36;"
  },
  BMD: {
    name: "Bermuda Dollar",
    symbol: "&#36;"
  },
  BOB: {
    name: "Bolivia Boliviano",
    symbol: "&#36;&#98;"
  },
  BAM: {
    name: "Bosnia and Herzegovina Convertible Marka",
    symbol: "&#75;&#77;"
  },
  BWP: {
    name: "Botswana Pula",
    symbol: "&#80;"
  },
  BGN: {
    name: "Bulgaria Lev",
    symbol: "&#1083;&#1074;"
  },
  BRL: {
    name: "Brazil Real",
    symbol: "&#82;&#36;"
  },
  BND: {
    name: "Brunei Darussalam Dollar",
    symbol: "&#36;"
  },
  KHR: {
    name: "Cambodia Riel",
    symbol: "&#6107;"
  },
  CAD: {
    name: "Canada Dollar",
    symbol: "&#36;"
  },
  KYD: {
    name: "Cayman Islands Dollar",
    symbol: "&#36;"
  },
  CLP: {
    name: "Chile Peso",
    symbol: "&#36;"
  },
  CNY: {
    name: "China Yuan Renminbi",
    symbol: "&#165;"
  },
  COP: {
    name: "Colombia Peso",
    symbol: "&#36;"
  },
  CRC: {
    name: "Costa Rica Colon",
    symbol: "&#8353;"
  },
  HRK: {
    name: "Croatia Kuna",
    symbol: "&#107;&#110;"
  },
  CUP: {
    name: "Cuba Peso",
    symbol: "&#8369;"
  },
  CZK: {
    name: "Czech Republic Koruna",
    symbol: "&#75;&#269;"
  },
  DKK: {
    name: "Denmark Krone",
    symbol: "&#107;&#114;"
  },
  DOP: {
    name: "Dominican Republic Peso",
    symbol: "&#82;&#68;&#36;"
  },
  XCD: {
    name: "East Caribbean Dollar",
    symbol: "&#36;"
  },
  EGP: {
    name: "Egypt Pound",
    symbol: "&#163;"
  },
  SVC: {
    name: "El Salvador Colon",
    symbol: "&#36;"
  },
  EEK: {
    name: "Estonia Kroon",
    symbol: "&#107;&#114;"
  },
  EUR: {
    name: "Euro",
    symbol: "&#8364;"
  },
  FKP: {
    name: "Falkland Islands (Malvinas) Pound",
    symbol: "&#163;"
  },
  FJD: {
    name: "Fiji Dollar",
    symbol: "&#36;"
  },
  GHC: {
    name: "Ghana Cedis",
    symbol: "&#162;"
  },
  GIP: {
    name: "Gibraltar Pound",
    symbol: "&#163;"
  },
  GTQ: {
    name: "Guatemala Quetzal",
    symbol: "&#81;"
  },
  GGP: {
    name: "Guernsey Pound",
    symbol: "&#163;"
  },
  GYD: {
    name: "Guyana Dollar",
    symbol: "&#36;"
  },
  HNL: {
    name: "Honduras Lempira",
    symbol: "&#76;"
  },
  HKD: {
    name: "Hong Kong Dollar",
    symbol: "&#36;"
  },
  HUF: {
    name: "Hungary Forint",
    symbol: "&#70;&#116;"
  },
  ISK: {
    name: "Iceland Krona",
    symbol: "&#107;&#114;"
  },
  INR: {
    name: "India Rupee",
    symbol: "&#8377;"
  },
  IDR: {
    name: "Indonesia Rupiah",
    symbol: "&#82;&#112;"
  },
  IRR: {
    name: "Iran Rial",
    symbol: "&#65020;"
  },
  IMP: {
    name: "Isle of Man Pound",
    symbol: "&#163;"
  },
  ILS: {
    name: "Israel Shekel",
    symbol: "&#8362;"
  },
  JMD: {
    name: "Jamaica Dollar",
    symbol: "&#74;&#36;"
  },
  JPY: {
    name: "Japan Yen",
    symbol: "&#165;"
  },
  JEP: {
    name: "Jersey Pound",
    symbol: "&#163;"
  },
  KZT: {
    name: "Kazakhstan Tenge",
    symbol: "&#1083;&#1074;"
  },
  KPW: {
    name: "Korea (North) Won",
    symbol: "&#8361;"
  },
  KRW: {
    name: "Korea (South) Won",
    symbol: "&#8361;"
  },
  KGS: {
    name: "Kyrgyzstan Som",
    symbol: "&#1083;&#1074;"
  },
  LAK: {
    name: "Laos Kip",
    symbol: "&#8365;"
  },
  LVL: {
    name: "Latvia Lat",
    symbol: "&#76;&#115;"
  },
  LBP: {
    name: "Lebanon Pound",
    symbol: "&#163;"
  },
  LRD: {
    name: "Liberia Dollar",
    symbol: "&#36;"
  },
  LTL: {
    name: "Lithuania Litas",
    symbol: "&#76;&#116;"
  },
  MKD: {
    name: "Macedonia Denar",
    symbol: "&#1076;&#1077;&#1085;"
  },
  MYR: {
    name: "Malaysia Ringgit",
    symbol: "&#82;&#77;"
  },
  MUR: {
    name: "Mauritius Rupee",
    symbol: "&#8360;"
  },
  MXN: {
    name: "Mexico Peso",
    symbol: "&#36;"
  },
  MNT: {
    name: "Mongolia Tughrik",
    symbol: "&#8366;"
  },
  MZN: {
    name: "Mozambique Metical",
    symbol: "&#77;&#84;"
  },
  NAD: {
    name: "Namibia Dollar",
    symbol: "&#36;"
  },
  NPR: {
    name: "Nepal Rupee",
    symbol: "&#8360;"
  },
  ANG: {
    name: "Netherlands Antilles Guilder",
    symbol: "&#402;"
  },
  NZD: {
    name: "New Zealand Dollar",
    symbol: "&#36;"
  },
  NIO: {
    name: "Nicaragua Cordoba",
    symbol: "&#67;&#36;"
  },
  NGN: {
    name: "Nigeria Naira",
    symbol: "&#8358;"
  },
  NOK: {
    name: "Norway Krone",
    symbol: "&#107;&#114;"
  },
  OMR: {
    name: "Oman Rial",
    symbol: "&#65020;"
  },
  PKR: {
    name: "Pakistan Rupee",
    symbol: "&#8360;"
  },
  PAB: {
    name: "Panama Balboa",
    symbol: "&#66;&#47;&#46;"
  },
  PYG: {
    name: "Paraguay Guarani",
    symbol: "&#71;&#115;"
  },
  PEN: {
    name: "Peru Nuevo Sol",
    symbol: "&#83;&#47;&#46;"
  },
  PHP: {
    name: "Philippines Peso",
    symbol: "&#8369;"
  },
  PLN: {
    name: "Poland Zloty",
    symbol: "&#122;&#322;"
  },
  QAR: {
    name: "Qatar Riyal",
    symbol: "&#65020;"
  },
  RON: {
    name: "Romania New Leu",
    symbol: "&#108;&#101;&#105;"
  },
  RUB: {
    name: "Russia Ruble",
    symbol: "&#1088;&#1091;&#1073;"
  },
  SHP: {
    name: "Saint Helena Pound",
    symbol: "&#163;"
  },
  SAR: {
    name: "Saudi Arabia Riyal",
    symbol: "&#65020;"
  },
  RSD: {
    name: "Serbia Dinar",
    symbol: "&#1044;&#1080;&#1085;&#46;"
  },
  SCR: {
    name: "Seychelles Rupee",
    symbol: "&#8360;"
  },
  SGD: {
    name: "Singapore Dollar",
    symbol: "&#36;"
  },
  SBD: {
    name: "Solomon Islands Dollar",
    symbol: "&#36;"
  },
  SOS: {
    name: "Somalia Shilling",
    symbol: "&#83;"
  },
  ZAR: {
    name: "South Africa Rand",
    symbol: "&#82;"
  },
  LKR: {
    name: "Sri Lanka Rupee",
    symbol: "&#8360;"
  },
  SEK: {
    name: "Sweden Krona",
    symbol: "&#107;&#114;"
  },
  CHF: {
    name: "Switzerland Franc",
    symbol: "&#67;&#72;&#70;"
  },
  SRD: {
    name: "Suriname Dollar",
    symbol: "&#36;"
  },
  SYP: {
    name: "Syria Pound",
    symbol: "&#163;"
  },
  TWD: {
    name: "Taiwan New Dollar",
    symbol: "&#78;&#84;&#36;"
  },
  THB: {
    name: "Thailand Baht",
    symbol: "&#3647;"
  },
  TTD: {
    name: "Trinidad and Tobago Dollar",
    symbol: "&#84;&#84;&#36;"
  },
  TRY: {
    name: "Turkey Lira",
    symbol: null
  },
  TRL: {
    name: "Turkey Lira",
    symbol: "&#8356;"
  },
  TVD: {
    name: "Tuvalu Dollar",
    symbol: "&#36;"
  },
  UAH: {
    name: "Ukraine Hryvna",
    symbol: "&#8372;"
  },
  GBP: {
    name: "United Kingdom Pound",
    symbol: "&#163;"
  },
  USD: {
    name: "United States Dollar",
    symbol: "&#36;"
  },
  UYU: {
    name: "Uruguay Peso",
    symbol: "&#36;&#85;"
  },
  UZS: {
    name: "Uzbekistan Som",
    symbol: "&#1083;&#1074;"
  },
  VEF: {
    name: "Venezuela Bolivar",
    symbol: "&#66;&#115;"
  },
  VND: {
    name: "Viet Nam Dong",
    symbol: "&#8363;"
  },
  YER: {
    name: "Yemen Rial",
    symbol: "&#65020;"
  },
  ZWD: {
    name: "Zimbabwe Dollar",
    symbol: "&#90;&#36;"
  }
};
