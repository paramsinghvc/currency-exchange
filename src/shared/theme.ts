const theme = {
  primaryColor: "#3A539B",
  borderRadius: "5px",
  borderColor: "#F2F2F2",
  subTextColor: "#979797",
  darkGrayTextColor: "#646464",
  highlightColor: "#425eb1"
};

export default theme;
