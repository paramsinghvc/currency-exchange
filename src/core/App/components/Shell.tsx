import React, { FC } from "react";
import styled from "@emotion/styled";

import Navbar from "./Navbar";
import useRedux from "@mollycule/redux-hook";
import { IRootState } from "shared/types";

const AppHolder = styled.main<{ blur?: boolean }>`
  background: #fff;
  height: 100%;
  min-height: 100vh;
  ${({ blur }) =>
    blur &&
    `
  filter: blur(6px);
  -webkit-filter: blur(6px);
  `}
`;

const Shell: FC<{ showBackButton?: boolean }> = ({ showBackButton, children }) => {
  const { currencyModalStatus } = useRedux<IRootState, Pick<IRootState["home"], "currencyModalStatus">>(
    state => ({
      currencyModalStatus: state.home.currencyModalStatus
    }),
    {}
  );

  return (
    <AppHolder blur={currencyModalStatus}>
      <Navbar showBackButton={showBackButton} />
      {children}
    </AppHolder>
  );
};

export default Shell;
