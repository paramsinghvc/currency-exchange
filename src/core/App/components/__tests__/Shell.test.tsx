import React from "react";
import Shell from "../Shell";
import { createStoreContext } from "@mollycule/redux-hook";
import { IRootState } from "shared/types";
import createStore from "core/utils/createStore";

const store = createStore();
const { Provider } = createStoreContext<IRootState>();
import renderer from "react-test-renderer";

const tree = renderer.create(
  <Provider store={store}>
    <Shell>
      <p>Hola</p>
    </Shell>
  </Provider>
);

it("should render the component", () => {
  expect(tree.toJSON()).toMatchSnapshot();
});
