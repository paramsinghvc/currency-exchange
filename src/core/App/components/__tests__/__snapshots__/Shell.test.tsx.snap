// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render the component 1`] = `
<main
  className="css-o8uaij"
>
  <header
    className="css-uf1ume"
  >
    <section
      className="css-m6ba0"
      role="button"
    >
      <span />
      <span />
      <span />
    </section>
  </header>
  <p>
    Hola
  </p>
</main>
`;
